src/go/rpk/pkg/config/params.go

```diff
@@ -295,9 +297,8 @@ func (p *Params) Load(fs afero.Fs) (*Config, error) {
 
 // Write writes loaded configuration parameters to redpanda.yaml.
 func (c *Config) Write(fs afero.Fs) (rerr error) {
-	cfgPath := c.loadedPath
-	if cfgPath == "" {
-		cfgPath = c.ConfigFile
+	if c.fileLocation == "" {
+		c.fileLocation = DefaultConfigPath
```

Comment:
Why would the file location still be empty here?

Also, I don't think this function should modify `c.fileLocation` -- the previous code modified a local copy of the variable. Basically, I wouldn't expect Write to have side effects on the input struct (the only reason it's a pointer is just to avoid copying the struct when calling the function). Is there benefit to setting `c.fileLocation`?

Reply:
True, `fileLocation` should NOT be empty at this point. Load handles all the cases. Thanks :+1:

Reply:
Might still be worth using the default location just in case as sanity checking 😅

Reply:
We would have empty fileLocation (and a SegFault) if we call cfg.Write without calling params.Load() first.

That's why I changed: https://github.com/redpanda-data/redpanda/pull/5709/files#diff-255d9a6d1d658202764d5ffd37eaf720c2c6eea3c6dfd4f7960191420415b397L514 and other tests.

I can put the check back on

Reply:
I'm ok with ensuring non-empty in Write